kahuna/public/js/search/query.js

```diff
@@ -36,7 +36,7 @@ query.controller('SearchQueryCtrl',
     // Boring - stops slashes being encoded in input - might be fixed with:
     // https://github.com/angular-ui/ui-router/issues/1759
     $scope.$watch(() => ctrl.filter.query, onValChange(newVal => {
-        ctrl.filter.query = decodeURIComponent(newVal);
+        ctrl.filter.query = angular.isDefined(newVal) ? decodeURIComponent(newVal) : '';
```

Comment:
I note that this 1. is a self-referencing observer/mutation (bit confusing and risk of endless loop) 2. changes `undefined` to `""` (might not matter?).